AEPMessaging/Sources/Messaging.swift

```diff
@@ -160,20 +161,27 @@ public class Messaging: NSObject, Extension {
     /// - Parameter event: an `Event` containing an in-app message definition in its data
     private func handleEdgePersonalizationNotification(_ event: Event) {
         // validate the event
-        guard event.isPersonalizationDecisionResponse, event.requestEventId == requestMessagesEventId else {
+        guard event.isPersonalizationDecisionResponse, event.requestEventId == messagesRequestEventId else {
             // either this isn't the type of response we are waiting for, or it's not a response for our request
             return
         }
-
-        guard let propositions = event.payload, !propositions.isEmpty, event.scope == appSurface else {
-            Log.debug(label: MessagingConstants.LOG_TAG, "Payload for in-app messages was empty. Clearing in-app messages from cache and persistence.")
-            rulesEngine.clearPropositions()                        
+        
+        // quick out if we have a scope (implying payload is not empty) and the scope doesn't match our app surface
+        if event.scope != nil && event.scope != appSurface {
+            Log.debug(label: MessagingConstants.LOG_TAG, "Ignoring response for personalization:decisions where the scope does not match the surface for in-app messages.")
             return
         }
-
-        rulesEngine.setPropositionsCache(propositions)
-        Log.trace(label: MessagingConstants.LOG_TAG, "Loading in-app message definition from network response.")
-        rulesEngine.loadPropositions(propositions)
+        
+        // if this is an event for a new request, purge cache and update lastProcessedRequestEventId
+        var clearExistingRules = false
+        if lastProcessedRequestEventId != event.requestEventId {
+            clearExistingRules = true            
+            lastProcessedRequestEventId = event.requestEventId
+        }
+                 
+        Log.trace(label: MessagingConstants.LOG_TAG, "Loading in-app message definitions from network response.")
+        rulesEngine.addPropositionsToCache(event.payload)
```

Comment:
I think [event.payload](https://github.com/adobe/aepsdk-messaging-ios/blob/00348ab14c41be3f0b23a231881d58f6b3dd6dc2/AEPMessaging/Sources/Event%2BMessaging.swift#L217) needs to filter just the prepositions relevant for the requested scope(s) before passing them to addPropositionsToCache